xcdat/regridder/regrid2.py

```diff
@@ -46,259 +46,170 @@ def __init__(self, input_grid: xr.Dataset, output_grid: xr.Dataset, **options: A
+    dst_lon_bnds,
+    input_grid: xr.Dataset,
+    output_grid: xr.Dataset,
+) -> xr.Dataset:
+    input_data_var = ds[data_var]
 
-    def _create_output_dataset(
-        self,
-        input_ds: xr.Dataset,
-        data_var: str,
-        output_data: np.ndarray,
-        axis_variable_name_map: Dict[str, str],
-        ordered_axis_names: List[str],
-    ) -> xr.Dataset:
-        """
-        Creates the output Dataset containing the new variable on the destination grid.
+    output_coords: dict[str, xr.DataArray] = {}
+    output_data_vars: dict[str, xr.DataArray] = {}
 
-        Parameters
-        ----------
-        input_ds : xr.Dataset
-            Input dataset containing coordinates and bounds for unmodified axes.
-        data_var : str
-            The name of the regridded variable.
-        output_data : np.ndarray
-            Output data array.
-        axis_variable_name_map : Dict[str, str]
-            Map of axis name e.g. ("X", "Y", etc) to variable name e.g. ("lon", "lat", etc).
-        ordered_axis_names : List[str]
-            List of axis names in the order observed for ``output_data``.
-
-        Returns
-        -------
-        xr.Dataset
-            Dataset containing the variable on the destination grid.
-        """
-        variable_axis_name_map = {y: x for x, y in axis_variable_name_map.items()}
+    dims = list(input_data_var.dims)
 
-        coords = {}
+    output_da = xr.DataArray(
+        output_data,
+        dims=dims,
+        coords=output_coords,
+        attrs=ds[data_var].attrs.copy(),
+        name=data_var,
+    )
 
-        # Grab coords and bounds from appropriate dataset.
-        for variable_name, axis_name in variable_axis_name_map.items():
-            if axis_name in ["X", "Y"]:
-                coords[variable_name] = self._output_grid[variable_name].copy()
-            else:
-                coords[variable_name] = input_ds[variable_name].copy()
+    output_data_vars[data_var] = output_da
 
-        output_da = xr.DataArray(
-            output_data,
-            dims=[axis_variable_name_map[x] for x in ordered_axis_names],
-            coords=coords,
-            attrs=input_ds[data_var].attrs.copy(),
-        )
+    output_ds = xr.Dataset(
+        output_data_vars,
+        attrs=input_grid.attrs.copy(),
+    )
 
-        data_vars = {data_var: output_da}
+    output_ds = _preserve_bounds(ds, output_grid, output_ds, ["X", "Y"])
 
-        return xr.Dataset(data_vars, attrs=input_ds.attrs.copy())
+    return output_ds
 
 
-def _map_latitude(src: xr.DataArray, dst: xr.DataArray) -> Tuple[List, List]:
+def _map_latitude(src: np.ndarray, dst: np.ndarray) -> Tuple[List, List]:
     """
     Map source to destination latitude.
 
     Parameters
     ----------
-    src : xr.DataArray
+    src : np.ndarray
         DataArray containing the source latitude bounds.
-    dst : xr.DataArray
+    dst : np.ndarray
         DataArray containing the destination latitude bounds.
```

Comment:
Update parameter description to `Array containing the destination latitude bounds.`

Reply:
Fixed in https://github.com/xCDAT/xcdat/pull/533/commits/96eaae21ebdf20fe51831bca24a8a892a6ed6cf0